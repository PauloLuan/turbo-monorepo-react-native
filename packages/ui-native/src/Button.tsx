import * as React from "react";
import { Button as RNButton } from "react-native";
import type { ButtonProps } from "react-native";

export const Button = ({ title = "Cool Native", ...props }: ButtonProps) => {
  return <RNButton title={title} {...props} />;
};
