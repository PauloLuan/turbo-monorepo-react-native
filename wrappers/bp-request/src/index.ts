import axios from "axios";

const get = async <P = null, T = any>(
  url: string,
  headers?: { [key: string]: string }
): Promise<T> => {
  const result = await axios.get<P, T>(url, {
    headers,
  });

  return result;
};

export { get };
