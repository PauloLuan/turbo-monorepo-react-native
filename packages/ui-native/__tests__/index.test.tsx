import "react-native";
import * as React from "react";
import { Button } from "../src/Button";
import renderer from "react-test-renderer";
describe("Test test", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
