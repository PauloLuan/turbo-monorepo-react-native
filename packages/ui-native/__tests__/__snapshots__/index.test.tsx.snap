// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test test renders correctly 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Cool Native   
    </Text>
  </View>
</View>
`;
